### PR TITLE
Implement missing features needeed for WebRTC Direct

### PR DESCRIPTION
**Pull request by altonen**
This commit introduces few minor changes to `str0m`:
  * allow user to set local ICE credentials
  * allow user to disable certificate fingerprint verification
  * allow user to fetch remote's certificate fingerprint

These changes allow implementing WebRTC Direct [1] using `str0m`

[1] https://github.com/libp2p/specs/blob/master/webrtc/webrtc-direct.md

---
Related to #166 

cc @thomaseizinger @xnorpx

### CODE REVIEW

File: src/change/direct.rs [hunk 1]
```diff
@@ -65,6 +70,11 @@ impl<'a> DirectApi<'a> {
         self.rtc.dtls.local_fingerprint().clone()
     }
 
+    /// Returns a reference to the local DTLS fingerprint used by this peer connection.
```
Comment (algesten) [inline src/change/direct.rs:73]: Fix this doc

File: src/lib.rs [hunk 2]
```diff
@@ -1610,6 +1615,20 @@ impl RtcConfig {
         self
     }
 
+    /// Set DTLS certification.
+    pub fn set_dtls_certification(mut self, dtls_cert: DtlsCert) -> Self {
+        self.dtls_cert = dtls_cert;
+        self
+    }
```
Comment (algesten) [inline src/lib.rs:1622]: Move this to after `dtls_cert` and rename it to `set_dtls_cert`.

File: src/lib.rs [hunk 3]
```diff
@@ -1610,6 +1615,20 @@ impl RtcConfig {
...
+
+    /// Toggle certificate fingerprint verification.
+    ///
+    /// By default the certificate fingerprint is verified.
+    pub fn set_certificate_fingerprint_verification(mut self, enabled: bool) -> Self {
+        self.certificate_fingerprint_verification = enabled;
+        self
+    }
```
Comment (algesten) [inline src/lib.rs:1630]: 1. For consistent naming, rename this to `set_fingerprint_verification` (in all places).
2. Make a getter `fingerprint_verification` that has a doc test which checks the default is `true`.

**Review by algesten — CHANGES_REQUESTED** [review]
Hi @altonen!

Thanks for this PR. There are a couple of small things to fix. However on an overall level, can you split this into 3 commits, one for each thing changing?

Also, we'd like to understand your use case. Disabling certificate verification is sometimes frowned upon in other contexts. What is the reason you need this? Is it actually a verification callback you are looking for?

**Comment by altonen** [comment]
Hi @algesten, I'll make the necessary changes.

W.r.t. to disabling certificate verification, quoting from the spec:

> **No need for trusted TLS certificates.** Enable browsers to connect to public server nodes without those server nodes providing a TLS certificate within the browser's trustchain. Note that we can not do this today with our Websocket transport as the browser requires the remote to have a trusted TLS certificate. Nor can we establish a plain TCP or QUIC connection from within a browser. We can establish a WebTransport connection from the browser (see [WebTransport specification](https://github.com/libp2p/specs/blob/master/webtransport)).

and
> A and B execute the DTLS handshake as part of the standard WebRTC connection establishment.
At this point B does not know the TLS certificate fingerprint of A. Thus B can not verify A's TLS certificate fingerprint during the DTLS handshake. Instead B needs to disable certificate fingerprint verification (see e.g. [Pion's disableCertificateFingerprintVerification option](https://github.com/pion/webrtc/blob/360b0f1745c7244850ed638f423cda716a81cedf/settingengine.go#L62)).

In case you're interested, here's the full process for establishing browser<->server connection: https://github.com/libp2p/specs/blob/master/webrtc/webrtc-direct.md#browser-to-public-server

This is needed for Polkadot so that light clients can connect to full nodes over WebRTC

File: src/change/direct.rs [hunk 4]
```diff
@@ -57,6 +57,11 @@ impl<'a> DirectApi<'a> {
         self.rtc.ice.set_remote_credentials(remote_ice_credentials);
     }
 
+    /// Sets the local ICE credentials.
+    pub fn set_local_ice_credentials(&mut self, local_ice_credentials: IceCreds) {
+        self.rtc.ice.set_local_credentials(local_ice_credentials);
+    }
+
```
[the remark this answers is not in the document]
Comment (altonen) [reply src/change/direct.rs:59]: Done

File: src/change/direct.rs (hunk 1 above)
Comment (altonen) [reply src/change/direct.rs]: Done

File: src/lib.rs (hunk 2 above)
Comment (altonen) [reply src/lib.rs:1631]: Done

File: src/lib.rs (hunk 3 above)
Comment (altonen) [reply src/lib.rs:1639]: Done

**Review by algesten — APPROVED** [review]
LGTM!

**Comment by algesten** [comment]
Thanks!

**Comment by altonen** [comment]
Cheers, any chance a new release could be cut some time in the near future?

**Comment by algesten** [comment]
Absolutely!

I haven't made progress on it in a couple of days, but I think we're ready: https://str0m.zulipchat.com/#narrow/stream/377845-general/topic/Release.200.2E2.2E0/near/387221067

**Comment by algesten** [comment]
0.2.0 is released: https://crates.io/crates/str0m/0.2.0

**Comment by altonen** [comment]
thanks!

**Comment by algesten** [comment]
Looking forward to hear how it works out for you!

**Pull request merged**